test/src/test/java/org/corfudb/runtime/clients/BaseClientTest.java

```diff
@@ -0,0 +1,67 @@
+package org.corfudb.runtime.clients;
+
+import io.netty.channel.nio.NioEventLoopGroup;
+import org.assertj.core.api.Assertions;
+import org.corfudb.protocols.wireprotocol.VersionInfo;
+import org.corfudb.runtime.RuntimeParameters;
+import org.corfudb.util.NodeLocator;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.UUID;
+
+@Ignore("Required only for integration Tests with a Corfu server running on port 9000")
+public class BaseClientTest {
+
+    private BaseClient baseClient;
+    private final int connTimeout = 10000000;
+
+    @Before
+    public void setup() {
+        NodeLocator nodeLocator = NodeLocator.parseString("localhost:9000");
+        UUID clientID = UUID.fromString("00000000-0000-0000-0000-000000000000");
+        RuntimeParameters runtimeParameters = new RuntimeParameters();
+        runtimeParameters.setClientId(clientID);
+        runtimeParameters.setCustomNettyChannelOptions(RuntimeParameters.DEFAULT_CHANNEL_OPTIONS);
+        runtimeParameters.setConnectionTimeout(Duration.ofHours(1));
+        runtimeParameters.setRequestTimeout(Duration.ofHours(1));
+        runtimeParameters.setIdleConnectionTimeout(connTimeout);
+        NettyClientRouter ncr = new NettyClientRouter(nodeLocator,
+                new NioEventLoopGroup(),
+                runtimeParameters);
+        this.baseClient = new BaseClient(ncr, 0L, clientID);
+    }
+
+    @Test
+    public void pingTest() {
+        Assertions.assertThat(baseClient.ping().join().booleanValue()).isTrue();
+    }
+
+    @Test
+    public void restartTest() {
+        Assertions.assertThat(baseClient.restart().join().booleanValue()).isTrue();
+    }
+
+    @Test
+    public void resetTest() {
+        Assertions.assertThat(baseClient.reset().join().booleanValue()).isTrue();
+    }
+
+    @Test
+    public void sealTest() {
+        final long NEW_EPOCH = 1L;
+        Assertions.assertThat(baseClient.sealRemoteServer(NEW_EPOCH).join().booleanValue()).isTrue();
+    }
+
+    @Test
+    public void versionTest() {
+        VersionInfo viCorfuMsg = baseClient.getVersionInfo().join();
+        VersionInfo viProtoMsg = baseClient.getVersionInfo().join();
+        Assertions.assertThat(viCorfuMsg.getJvmUsed()).isEqualTo(viProtoMsg.getJvmUsed());
+        Assertions.assertThat(viCorfuMsg.getStartupArgs()).isEqualTo(viProtoMsg.getStartupArgs());
+        Assertions.assertThat(viCorfuMsg.getVersion()).isEqualTo(viProtoMsg.getVersion());
+        Assertions.assertThat(viCorfuMsg.getNodeId()).isEqualTo(viProtoMsg.getNodeId());
+    }
```

Comment:
Note: This will just invoke the RPC twice, and does not actually compare the old implementation with the new.